### ISSUE TITLE
Controll staging production swap

### ISSUE THREAD

**Issue by talarari** [dotnet/orleans#3545]
The docs say if you deploy a staging silo and also use a minimum version placement strategy, then untill you direct traffic to the staging silo - no activations will be created there.

So basically you can run some tests on the staging silo and then direct all traffic to it (swap).

From what i understand the new silo will have no activations because of the minimum version placement strategy, meaning old clients will request v1 grains and none of those exist in the staging silo.

My question is, what happens if i only bump the version on one grain from v1 to v2 (since its the only one changed, lets say i added a method)?
The staging silo in this case will have v1 for all other grains, so they could start getting activations for v1 the second they the silo goes online.
This means i cant controll when the swap happens (which should be after i validate the new silo).

Is my understanding correct? Or am i missing something?

**Comment by benjaminpetit** [comment]
> From what i understand the new silo will have no activations because of the minimum version placement strategy, meaning old clients will request v1 grains and none of those exist in the staging silo.

Correct

> My question is, what happens if i only bump the version on one grain from v1 to v2 (since its the only one changed, lets say i added a method)?
The staging silo in this case will have v1 for all other grains, so they could start getting activations for v1 the second they the silo goes online.
This means i cant controll when the swap happens (which should be after i validate the new silo).

Let me rephrase so I am sure I understood:

In your cluster, your Silo 1 contains grains A(v1) and Grain B(v2). You deploy Silo 2 with A(v1) and B(v2). If the strategy use is "minimum version placement " then grains of type A could still be placed on Silo 1 and 2. SInce you didn't changed the version, Orleans consider this to be safe.

If you want to isolate the new silo for validation, you should bump the version of grain of type A too.
